From e43512b78594dc8c55927b56504020ed10d732d0 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Mon, 20 Nov 2023 13:59:09 +0100
Subject: [PATCH 0688/1141] dt-bindings: memory-controller: Add octal-bus
 bindings

OSPI interface can be used as a communication channel (up to 8 lines)
with a device which is not a memory device.
Add device tree bindings for the octal-bus support.

Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: I26eb291e07aa03d030e6ec04f195cc904636a959
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/340950
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../memory-controllers/st,stm32-omi.yaml      | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/Documentation/devicetree/bindings/memory-controllers/st,stm32-omi.yaml b/Documentation/devicetree/bindings/memory-controllers/st,stm32-omi.yaml
index 56142eb5c3a9..110b670578b3 100644
--- a/Documentation/devicetree/bindings/memory-controllers/st,stm32-omi.yaml
+++ b/Documentation/devicetree/bindings/memory-controllers/st,stm32-omi.yaml
@@ -101,6 +101,7 @@ patternProperties:
           - enum:
               - jedec,spi-nor
               - spi-nand
+              - st,octal-bus
 
     allOf:
       - if:
@@ -109,6 +110,7 @@ patternProperties:
               enum:
                 - jedec,spi-nor
                 - spi-nand
+                - st,octal-bus
 
         then:
           $ref: "/schemas/spi/spi-controller.yaml#"
@@ -141,6 +143,17 @@ patternProperties:
           required:
             - st,max-frequency
 
+      - if:
+          properties:
+            compatible:
+              contains:
+                enum:
+                  - st,octal-bus
+
+        then:
+          required:
+            - spi-max-frequency
+
 additionalProperties: false
 
 examples:
@@ -202,3 +215,32 @@ examples:
         st,tacc-ns = <48>;
       };
     };
+  - |
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
+    #include <dt-bindings/bus/stm32mp25_sys_bus.h>
+    #include <dt-bindings/clock/stm32mp25-clks.h>
+    #include <dt-bindings/reset/stm32mp25-resets.h>
+    memory-controller@40430000 {
+      compatible = "st,stm32mp25-omi";
+      reg = <0x40430000 0x400>;
+      memory-region = <&mm_ospi1>;
+      interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>;
+      dmas = <&hpdma 2 0x62 0x00003121 0x0>,
+             <&hpdma 2 0x42 0x00003112 0x0>;
+      dma-names = "tx", "rx";
+      st,syscfg-dlyb = <&syscfg 0x1000>;
+      clocks = <&scmi_clk CK_SCMI_OSPI1>;
+      resets = <&scmi_reset RST_SCMI_OSPI1>, <&scmi_reset RST_SCMI_OSPI1DLL>;
+      feature-domains = <&rifsc STM32MP25_RIFSC_OCTOSPI1_ID>;
+      power-domains = <&CLUSTER_PD>;
+      cs-gpios = <&gpiod 1 GPIO_ACTIVE_LOW>;
+
+      #address-cells = <1>;
+      #size-cells = <0>;
+
+      flash@0 {
+        compatible = "st,octal-bus";
+        reg = <0>;
+        spi-max-frequency = <50000000>;
+      };
+    };
-- 
2.39.2

